pkg/lightning/common/util.go

```diff
@@ -142,6 +143,11 @@ func (param *MySQLConnectParam) Connect() (*sql.DB, error) {
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
+	// The actual number of alive connections is controlled by the region concurrency
+	// The setting is required to avoid frequent connection creation and close
+	if db != nil {
```

Comment:
I think we don't need to check nilness. We have make sure it's not nil  when `err` is nil

Reply:
If I remove the nilness check, the unit test failed, not sure why https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftidb%2Fghpr_unit_test/detail/ghpr_unit_test/17153/pipeline

Reply:
Sorry we have a failpoint that returns (nil, nil)

Reply:
should I get back the check `if db != nil `?

Reply:
yes, sorry for my previous comment